src/librustc_codegen_llvm/metadata.rs

```diff
@@ -67,10 +66,16 @@ fn search_meta_section<'a>(of: &'a ObjectFile,
     unsafe {
         let si = mk_section_iter(of.llof);
         while llvm::LLVMIsSectionIteratorAtEnd(of.llof, si.llsi) == False {
-            let mut name_buf = ptr::null();
+            let mut name_buf = None;
             let name_len = llvm::LLVMRustGetSectionName(si.llsi, &mut name_buf);
-            let name = slice::from_raw_parts(name_buf as *const u8, name_len as usize).to_vec();
-            let name = String::from_utf8(name).unwrap();
+            let name = name_buf.map_or(
```

Comment:
For the purposes of this code, sections without name can be skipped entirely. I'm also wondering why this code is creating a `String` rather than a `str`, it does not seem necessary.

Reply:
Are you saying I should change something? I literally have no idea what the code here is doing, I'm afraid.